### PR TITLE
Add error bars to plots for #944

### PR DESCRIPTION
**Pull request by cmcantalupo**
Clean up plotting code considerably (Thanks @bgeltz) and switched to pandas.

History could use some squashing.

### CODE REVIEW

File: test_integration/test_ee_short_region_slop.py [hunk 1]
```diff
@@ -187,122 +192,133 @@ def tearDownClass(cls):
...
-        min_energy -= 0.05 * delta_energy
-        max_energy += 0.05 * delta_energy
-
-        all_runtime = list(scaling_data_fixed[column_idx(cols, 'runtime')])
-        all_runtime.extend(scaling_data_dynamic[column_idx(cols, 'runtime')])
-        all_runtime.extend(timed_data_fixed[column_idx(cols, 'runtime')])
-        all_runtime.extend(timed_data_dynamic[column_idx(cols, 'runtime')])
-        min_runtime = min(all_runtime)
-        max_runtime = max(all_runtime)
+        is_first_pass = True
```
Comment (cmcantalupo) [inline test_integration/test_ee_short_region_slop.py]: garbage

File: test_integration/test_ee_short_region_slop.py [hunk 2]
```diff
@@ -187,122 +192,133 @@ def tearDownClass(cls):
...
+        for trial_idx in range(self._num_trial):
+            try:
+                report_path = '{}.{}'.format(self._report_path_fixed, trial_idx)
+                report_fixed = geopmpy.io.RawReport(report_path)
+                report_path = '{}.{}'.format(self._report_path_dynamic, trial_idx)
+                report_dynamic = geopmpy.io.RawReport(report_path)
+                data = data.append(extract_data(report_fixed, trial_idx))
+                data = data.append(extract_data(report_dynamic, trial_idx))
+            except:
+                if trial_idx == 0:
```
Comment (cmcantalupo) [inline test_integration/test_ee_short_region_slop.py:221]: Print warning that some reports were missing

File: test_integration/test_ee_short_region_slop.py [hunk 3]
```diff
@@ -187,122 +192,133 @@ def tearDownClass(cls):
...
+    scaling_data['trial-idx'] = trial_idx
+    timed_data['trial-idx'] = trial_idx
+    return scaling_data.append(timed_data)
+
+
+def plot_data(data, region_type, xaxis, yaxis):
+    for policy_type in ('fixed', 'dynamic'):
+        idx = pandas.IndexSlice
+        prof_name = 'ee_short_region_slop-{}'.format(policy_type)
+        selected_data = data.loc[idx[prof_name, region_type], [xaxis, yaxis]]
```
Comment (cmcantalupo) [inline test_integration/test_ee_short_region_slop.py]: look into xs method.

File: test_integration/test_ee_short_region_slop.py [hunk 4]
```diff
@@ -187,122 +192,133 @@ def tearDownClass(cls):
...
                for rn in region_names
                if rn.startswith(key)]
-    result = []
-    for rr in regions:
-         sample = []
-         for cc in cols:
+    result = {}
+    sample = []
+    for (rn, rr) in regions:
+        sample.append(rn.split('_')[0])
```
Comment (cmcantalupo) [inline test_integration/test_ee_short_region_slop.py:322]: Needs a comment

**Review by bgeltz — DISMISSED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**